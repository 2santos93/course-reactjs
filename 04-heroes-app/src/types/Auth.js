export const AUTH = {
  LOGIN: 'Login',
  LOGOUT: 'Logout'
}