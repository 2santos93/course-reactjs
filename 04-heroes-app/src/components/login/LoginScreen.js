import React, { useContext } from 'react'
import { AuthContext } from '../../auth/AuthContext'
import { AUTH } from '../../types/Auth';

export const LoginScreen = ({history}) => {

  const {dispatch} = useContext(AuthContext);

  const loginHandler = () => {
    // history.push('/marvel');
    const loginAction = {
      type: AUTH.LOGIN,
      payload: {
        name: 'nelson'
      }
    }
    
    dispatch(loginAction);
    
    history.replace('/');
  }

  return (
    <>
      <h1>Login</h1>
      <hr />

      <button
        className='btn btn-primary'
        onClick={loginHandler}
      >
        Login
      </button>
    </>
  )
}
