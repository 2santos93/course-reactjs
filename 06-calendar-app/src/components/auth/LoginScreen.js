import React from 'react';
import './login.css';
import { useDispatch } from 'react-redux';
import { startLogin, startRegister } from '../../actions/auth';

export const LoginScreen = () => {
  const dispatch = useDispatch();

  const onLoginSubmitHandler = (e) => {
    e.preventDefault();
    const email = document.getElementById('lemail');
    const password = document.getElementById('lpassword');
    const emailValue = email.value;
    const passwordValue = password.value;

    if (emailValue.trim().length < 2 || passwordValue.trim().length < 6)
      return alert('error in login form');

    dispatch(startLogin(emailValue, passwordValue));
    email.value = '';
    password.value = '';
  };

  const onSubmitRegisterHandler = (e) => {
    e.preventDefault();
    const email = document.getElementById('Remail');
    const name = document.getElementById('Rname');
    const password = document.getElementById('Rpassword');
    const repeatPassword = document.getElementById('RrepeatPassword');

    const emailValue = email.value;
    const nameValue = name.value;
    const passwordValue = password.value;
    const repeatPasswordValue = repeatPassword.value;

    if (
      emailValue.trim().length < 2 ||
      passwordValue.trim().length < 6 ||
      nameValue.trim().length < 2 ||
      repeatPasswordValue !== passwordValue
    )
      return alert('error in register form');

    dispatch(startRegister(emailValue, passwordValue, nameValue));

    email.value = '';
    name.value = '';
    password.value = '';
    repeatPassword.value = '';
  };

  return (
    <div className="container login-container">
      <div className="row">
        <div className="col-md-6 login-form-1">
          <h3>Login</h3>
          <form onSubmit={onLoginSubmitHandler}>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Email"
                name="email"
                id="lemail"
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Password"
                name="password"
                id="lpassword"
              />
            </div>
            <div className="form-group">
              <input type="submit" className="btnSubmit" value="Login" />
            </div>
          </form>
        </div>

        <div className="col-md-6 login-form-2">
          <h3>Register</h3>
          <form onSubmit={onSubmitRegisterHandler}>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Name"
                id="Rname"
              />
            </div>
            <div className="form-group">
              <input
                type="email"
                className="form-control"
                placeholder="Email"
                id="Remail"
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Password"
                id="Rpassword"
              />
            </div>

            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Repeat password"
                name="repeatPassword"
                id="RrepeatPassword"
              />
            </div>

            <div className="form-group">
              <input type="submit" className="btnSubmit" value="Register" />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};
