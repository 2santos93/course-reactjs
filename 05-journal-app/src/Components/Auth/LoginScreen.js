import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { beforeGoogleLogin, beforeLogin } from "../../actions/auth";
import { useForm } from "./../../hooks/useForm";

export const LoginScreen = () => {
  const [formvalues, handleInputChange] = useForm();
  const {loading} = useSelector(state => state.ui)
  const dispatch = useDispatch();
  const { email, password } = formvalues;

  const onSubmitHandler = (e) => {
    e.preventDefault();
    dispatch(beforeLogin(email, password));
  };

  const onSubmitHandlerGoogle = () => {
    dispatch(beforeGoogleLogin());
  };

  return (
    <div>
      <h1 className="auth__header-login">Login</h1>
      <form onSubmit={onSubmitHandler} >
        <input
          type="email"
          name="email"
          className="auth__input"
          placeholder="Email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />
        <input
          type="password"
          className="auth__input"
          placeholder="Password"
          value={password}
          name="password"
          onChange={handleInputChange}
        />
        <button type="submit" className="auth__button" disabled={loading} >
          Login
        </button>
      </form>

      <div className="google-btn" onClick={onSubmitHandlerGoogle}>
        <div className="google-icon-wrapper">
          <img
            className="google-icon"
            src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
            alt="google button"
          />
        </div>
        <p className="btn-text">
          <b>Sign in with google</b>
        </p>
      </div>

      <div className="auth__footer">
        <Link to="/auth/register">Create new account</Link>
      </div>
    </div>
  );
};
