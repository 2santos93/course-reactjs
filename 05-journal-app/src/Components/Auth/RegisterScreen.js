import React from "react";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from 'react-redux';
import validator from 'validator';

import { useForm } from "./../../hooks/useForm";
import { removeErrorAction, setErrorAction } from "../../actions/ui";
import { beforeRegister } from "../../actions/auth";

export const RegisterScreen = () => {
  const [formValues, handleInputChange] = useForm({});
  const dispatch = useDispatch();
  const {errorMsg} = useSelector(state => state.ui);

  const { email='', name='', password='', confirmPassword='' } = formValues;

  const registerHandler = (e) => {
    e.preventDefault();

    if(!isFormValid()) return dispatch(setErrorAction());

    dispatch(removeErrorAction());
    dispatch(beforeRegister(email, password, name));

  };

  const isFormValid = () => {

    if(
      !validator.isEmail(email+'') ||
      validator.isEmpty(name+'') ||
      validator.isEmpty(password+'') ||
      // !validator.isStrongPassword(password+'', [{minLength: 5}]) || doesnt work
      password.trim().length < 5 ||
      !validator.equals(password+'', confirmPassword+'')
    ) return false;

    return true;
  }

  return (
    <div>
      <h1 className="auth__header-login">Register</h1>
      <form action="">

          {
          errorMsg && (
            <div className="error mt-1">
              {errorMsg}
            </div>
          )
          }

        <input
          type="text"
          name="name"
          value={name}
          className="auth__input"
          placeholder="Name"
          autoComplete="off"
          onChange={handleInputChange}
        />
        <input
          onChange={handleInputChange}
          value={email}
          type="email"
          name="email"
          className="auth__input"
          placeholder="Email"
          autoComplete="off"
        />
        <input
          onChange={handleInputChange}
          type="password"
          name="password"
          value={password}
          className="auth__input"
          placeholder="Password"
        />
        <input
          onChange={handleInputChange}
          type="password"
          value={confirmPassword}
          name="confirmPassword"
          className="auth__input"
          placeholder="Confirm Password"
        />
        <button 
          type="submit" 
          className="auth__button"
          onClick={registerHandler}
        >
          Register
        </button>
      </form>

      <div className="auth__footer">
        <Link to="/auth/login">Already have an account?</Link>
      </div>
    </div>
  );
};
