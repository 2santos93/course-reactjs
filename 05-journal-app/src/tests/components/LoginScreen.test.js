import { mount } from "enzyme";
import { LoginScreen } from "./../../Components/Auth/LoginScreen";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import { MemoryRouter } from "react-router";
import { act } from "@testing-library/react";

describe("<LoginComponent /> test", () => {
  const uid = "TESTING";
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const initState = {
    auth: { uid },
    notes: { notes: [], active: {} },
    ui: { loading: false, errorMsg: null },
  };

  const store = mockStore(initState);

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <LoginScreen />
      </MemoryRouter>
    </Provider>
  );

  test("should render ok", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should dispatch google login action", () => {

    wrapper.find(".google-btn").prop('onClick')();

    const actions = store.getActions();

    console.log(actions);
  });
});
