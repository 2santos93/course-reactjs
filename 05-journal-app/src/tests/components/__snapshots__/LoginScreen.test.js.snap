// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginComponent /> test should render ok 1`] = `
<div>
  <h1
    className="auth__header-login"
  >
    Login
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <button
      className="auth__button"
      disabled={false}
      type="submit"
    >
      Login
    </button>
  </form>
  <div
    className="google-btn"
    onClick={[Function]}
  >
    <div
      className="google-icon-wrapper"
    >
      <img
        alt="google button"
        className="google-icon"
        src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
      />
    </div>
    <p
      className="btn-text"
    >
      <b>
        Sign in with google
      </b>
    </p>
  </div>
  <div
    className="auth__footer"
  >
    <a
      href="/auth/register"
      onClick={[Function]}
    >
      Create new account
    </a>
  </div>
</div>
`;
